### ISSUE TITLE
[fread] does not recognize numbers written in European style

### ISSUE THREAD

**Issue by GegznaV** [Rdatatable/data.table#2430]
In the following example, function `fread` fails to recognize numbers written in European style (where decimal separator is comma):

```r
data(npk)
write.table(npk, "table.txt", row.names = FALSE, dec = ",", quote = FALSE)

str(data.table::fread("table.txt", dec = ","))
```
Result:
```
Classes ‘data.table’ and 'data.frame':	24 obs. of  5 variables:
 $ block: int  1 1 1 1 2 2 2 2 3 3 ...
 $ N    : int  0 1 0 1 1 1 0 0 0 1 ...
 $ P    : int  1 1 0 0 0 1 0 1 1 1 ...
 $ K    : int  1 0 0 1 0 1 1 0 0 1 ...
 $ yield: chr  "49,5" "62,8" "46,8" "57" ...
 - attr(*, ".internal.selfref")=<externalptr> 
```

`yield` is expected to be numeric.

**Comment by etienne-s** [comment]
It works for me:
```r
str(data.table::fread("table.txt", dec = ","))
# Classes ‘data.table’ and 'data.frame':	24 obs. of  5 variables:
#  $ block: int  1 1 1 1 2 2 2 2 3 3 ...
#  $ N    : int  0 1 0 1 1 1 0 0 0 1 ...
#  $ P    : int  1 1 0 0 0 1 0 1 1 1 ...
#  $ K    : int  1 0 0 1 0 1 1 0 0 1 ...
#  $ yield: num  49.5 62.8 46.8 57 59.8 58.5 55.5 56 62.8 55.8 ...
#  - attr(*, ".internal.selfref")=<externalptr> 
```

Could you provide your `sessionInfo()`?

**Comment by st-pasha** [comment]
@etienne-s The most recent version of `fread` treats 0/1 column as logical. This is what I'm getting:
```
> str(data.table::fread("table.txt", dec = ","))
Classes ‘data.table’ and 'data.frame':	24 obs. of  5 variables:
 $ block: int  1 1 1 1 2 2 2 2 3 3 ...
 $ N    : logi  FALSE TRUE FALSE TRUE TRUE TRUE ...
 $ P    : logi  TRUE TRUE FALSE FALSE FALSE TRUE ...
 $ K    : logi  TRUE FALSE FALSE TRUE FALSE TRUE ...
 $ yield: num  49.5 62.8 46.8 57 59.8 58.5 55.5 56 62.8 55.8 ...
 - attr(*, ".internal.selfref")=<externalptr> 
```
The `yield` is still numeric. At the same time `fread` is currently unable to automatically deduce that `sep=","` should be used (and as far as I understand it is not in the roadmap either).

**Comment by mattdowle** [comment]
Automatic detection of `dec=','` could be in the road map if enough people show enough interest.  We just need a bit of encouragement.  Filed a new enhancement request here : https://github.com/Rdatatable/data.table/issues/2431
Btw, there was an enhancement for `dec` in dev in NEWS : 

> dec=',' is now implemented directly so there is no dependency on locale. The options datatable.fread.dec.experiment and datatable.fread.dec.locale have been removed.

Looks like setting `dec=','` is working fine so closing.  Please reopen if I misunderstood.

**Comment by GegznaV** [comment]
I installed and loaded package `rio`, then tried the code I provided in my question.

```r
library("rio")
data(npk)
write.table(npk, "table.txt", row.names = FALSE, dec = ",", quote = FALSE)

str(data.table::fread("table.txt", dec = ","))
```

And I still can reproduce the error. Before loading `rio` the `dec =","` worked fine.

Can you reproduce the error by loading the `rio`?

Is here any "sectrt" `data.table` parameters, that chan be modified by other packages?

**Comment by etienne-s** [comment]
I've installed and loaded `rio` and I still cannot reproduce. Strange.

**Comment by GegznaV** [comment]
Session info, when `fread` **works:**
```r
devtools::session_info()
Session info -----------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, mingw32             
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  English_United States.1252  
 tz       Europe/Helsinki             
 date     2017-10-21                  

Packages ---------------------------------------------------------------------------------------------------------
 package    * version  date       source                          
 base       * 3.4.2    2017-09-28 local                           
 compiler     3.4.2    2017-09-28 local                           
 data.table   1.10.4-2 2017-10-12 CRAN (R 3.4.2)                  
 datasets   * 3.4.2    2017-09-28 local                           
 devtools     1.13.3   2017-08-02 CRAN (R 3.4.2)                  
 digest       0.6.12   2017-01-27 CRAN (R 3.4.1)                  
 graphics   * 3.4.2    2017-09-28 local                           
 grDevices  * 3.4.2    2017-09-28 local                           
 memoise      1.1.0    2017-04-21 CRAN (R 3.4.1)                  
 methods    * 3.4.2    2017-09-28 local                           
 stats      * 3.4.2    2017-09-28 local                           
 tools        3.4.2    2017-09-28 local                           
 utils      * 3.4.2    2017-09-28 local                           
 withr        2.0.0    2017-10-17 Github (jimhester/withr@a43df66)
 yaml         2.1.14   2016-11-12 CRAN (R 3.4.1)      
```

Session info, when `fread` **breaks:**
```
> library("rio")
> devtools::session_info()
Session info -----------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.2 (2017-09-28)
 system   x86_64, mingw32             
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  English_United States.1252  
 tz       Europe/Helsinki             
 date     2017-10-21                  

Packages ---------------------------------------------------------------------------------------------------------
 package    * version    date       source                          
 base       * 3.4.2      2017-09-28 local                           
 cellranger   1.1.0      2016-07-27 CRAN (R 3.4.1)                  
 compiler     3.4.2      2017-09-28 local                           
 curl         3.0        2017-10-06 CRAN (R 3.4.2)                  
 data.table   1.10.4-2   2017-10-12 CRAN (R 3.4.2)                  
 datasets   * 3.4.2      2017-09-28 local                           
 devtools     1.13.3     2017-08-02 CRAN (R 3.4.2)                  
 digest       0.6.12     2017-01-27 CRAN (R 3.4.1)                  
 forcats      0.2.0      2017-01-23 CRAN (R 3.4.1)                  
 foreign      0.8-69     2017-06-22 CRAN (R 3.4.2)                  
 graphics   * 3.4.2      2017-09-28 local                           
 grDevices  * 3.4.2      2017-09-28 local                           
 haven        1.1.0      2017-07-09 CRAN (R 3.4.1)                  
 magrittr     1.5        2014-11-22 CRAN (R 3.4.1)                  
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.1)                  
 methods    * 3.4.2      2017-09-28 local                           
 openxlsx     4.0.17     2017-03-23 CRAN (R 3.4.2)                  
 Rcpp         0.12.13    2017-09-28 CRAN (R 3.4.2)                  
 readxl       1.0.0      2017-04-18 CRAN (R 3.4.1)                  
 rio        * 0.5.5      2017-06-18 CRAN (R 3.4.2)                  
 rlang        0.1.2.9000 2017-09-21 Github (tidyverse/rlang@99105b6)
 rstudioapi   0.7        2017-09-07 CRAN (R 3.4.1)                  
 stats      * 3.4.2      2017-09-28 local                           
 tibble       1.3.4      2017-08-22 CRAN (R 3.4.1)                  
 tools        3.4.2      2017-09-28 local                           
 utils      * 3.4.2      2017-09-28 local                           
 withr        2.0.0      2017-10-17 Github (jimhester/withr@a43df66)
 yaml         2.1.14     2016-11-12 CRAN (R 3.4.1)   
```

**Comment by mattdowle** [comment]
@GegznaV Just seen these extra comments using 1.10.4-2. As I wrote higher up, `dec=` is much better implemented in dev (currently 1.10.5) and I believe this problem is now solved. Please open a new issue if it is not.
